### PR TITLE
kbfsgit: add memory profiles for each stage when verbosity >= 3

### PR DESCRIPTION
**Pull request by strib**
Also re-vendor in go-git to include some minor master performance improvements, and a new status stage.

Issue: KBFS-2404

### CODE REVIEW

File: kbfsgit/runner.go [hunk 1]
```diff
@@ -450,6 +453,17 @@ func (r *runner) processGogitStatus(
 	lastByteCount := 0
 	for update := range statusChan {
 		if update.Stage != currStage {
+			if r.verbosity >= 3 {
+				f, err := os.Create(filepath.Join(
+					os.TempDir(), fmt.Sprintf("memprof.%d.prof", update.Stage)))
```
Comment (jzila) [inline kbfsgit/runner.go]: Can you output the location of this file as part of the status dump?

Comment (strib) [reply kbfsgit/runner.go]: By "status dump" you mean the errput data that's being printed as part of the `git` execution to the terminal?  Sure, will do.

Comment (jzila) [reply kbfsgit/runner.go]: Yep, thanks.

File: kbfsgit/runner.go [hunk 2]
```diff
@@ -450,6 +453,17 @@ func (r *runner) processGogitStatus(
...
+				if err != nil {
+					r.log.CDebugf(ctx, err.Error())
+				} else {
```
Comment (patrickxb) [inline kbfsgit/runner.go]: might want to try `runtime.GC()` here, otherwise profiles could be skewed by garbage.

Comment (strib) [reply kbfsgit/runner.go]: Thanks, done.

**Pull request merged**